### PR TITLE
ClinicalTrials Harvester & Normalizer

### PR DESCRIPTION
**Pull request by caseyrollins**

### CODE REVIEW

**Comment by caseyrollins** [comment]
@chrisseto 🐠

File: providers/gov/clinicaltrials/harvester.py [hunk 1]
```diff
@@ -1,5 +1,69 @@
...
+        end_year = end_date.strftime('%y')
+        end_month = end_date.strftime('%m')
+
+        start_day = start_date.strftime('%d')
+        start_year = start_date.strftime('%y')
+        start_month = start_date.strftime('%m')
+
+        return self.fetch_records(furl(self.url).set(query_params={
+            'displayxml': 'true',
+            'lup_s': '{}/{}/{}'.format(start_month, start_day, start_year),
```
Comment (chrisseto) [inline providers/gov/clinicaltrials/harvester.py]: `.strftime('%m/%d/%y')`

File: providers/gov/clinicaltrials/harvester.py [hunk 2]
```diff
@@ -1,5 +1,69 @@
...
+                }).url
+                for record in all_records_doc.xpath('//clinical_study')
+            ]
+
+            logger.info("There are {} record urls to harvest - this may take a while...".format(len(record_urls)))
+            for url in record_urls:
+                try:
+                    record_resp = self.requests.get(url)
+                except self.requests.exceptions.ConnectionError as e:
+                    logger.info('Connection error: {}, wait a bit...'.format(e))
```
Comment (chrisseto) [inline providers/gov/clinicaltrials/harvester.py]: Use `logger.warning` here

Comment (chrisseto) [inline providers/gov/clinicaltrials/harvester.py]: Rather than continue make the same request again, otherwise we'll end up losing that data.

**Comment by caseyrollins** [comment]
@chrisseto 🐙

File: providers/gov/clinicaltrials/normalizer.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+from share.normalize import *
+
+
+class Venue(Parser):
+    name = Maybe(Maybe(ctx, 'facility'), 'name')
+
+
+class ThroughVenues(Parser):
+    pass
```
Comment (chrisseto) [inline providers/gov/clinicaltrials/normalizer.py]: `venue = Delegate(Venue, ctx)`?

File: providers/gov/clinicaltrials/normalizer.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+
+class Tag(Parser):
+    name = ctx
+
+
+class ThroughTags(Parser):
+    pass
```
Comment (chrisseto) [inline providers/gov/clinicaltrials/normalizer.py]: `tag = Delegate(Tag, ctx)`?

File: providers/gov/clinicaltrials/normalizer.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+
+class Email(Parser):
+    email = ctx
+
+
+class PersonEmail(Parser):
+    pass
```
Comment (chrisseto) [inline providers/gov/clinicaltrials/normalizer.py]: `email = ...`?

**Comment by caseyrollins** [comment]
@chrisseto 👽

File: providers/gov/clinicaltrials/normalizer.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    given_name = Maybe(ctx, 'first_name')
+    family_name = Maybe(ctx, 'last_name')
+    additional_name = Maybe(ctx, 'middle_name')
+    emails = Map(Delegate(PersonEmail), Maybe(ctx, 'email'))
+    affiliations = Map(Delegate(Affiliation.using(entity=Delegate(Entity))), Maybe(ctx, 'affiliation'))
+
+
+class Contributor(Parser):
+    order_cited = ctx('index')
+    cited_name = ''
```
Comment (chrisseto) [inline providers/gov/clinicaltrials/normalizer.py]: Join names?

**Comment by chrisseto** [comment]
¯\_(ツ)_/¯ Maybe

**Pull request merged**